packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmReference.kt

```diff
@@ -73,7 +73,15 @@ public data class FrozenRealmReference(
     override val owner: BaseRealmImpl,
     override val dbPointer: FrozenRealmPointer,
     override val schemaMetadata: SchemaMetadata = CachedSchemaMetadata(dbPointer),
-) : RealmReference
+) : RealmReference {
+    init {
+        // realm_open/realm_freeze doesn't implicitly create a transaction which can cause the
+        // underlying core version to be cleaned up if the realm is advanced before any objects,
+        // queries, etc. triggers creation of the transaction. Thus, we need to force a transaction
+        // on any realm references to keep the version around for future operations.
+        RealmInterop.realm_begin_read(dbPointer)
```

Comment:
I thought obtaining a Frozen Realm from core will automatically pin the underlying version without the need of starting a read transaction ...

Reply:
Yes, thought that too. It also sounds like core team would agree that this would be the correct approach, but historically some usage patterns have called for not doing it ... and it seems less risky for us to just fix it locally until the approach is revisited in a broader scope.